pkg/manifestclient/group_resource_discovery.go

```diff
@@ -4,14 +4,58 @@ import (
 	"errors"
 	"fmt"
 	"io/fs"
+	"k8s.io/apimachinery/pkg/util/json"
+	"os"
 	"path/filepath"
+	"sigs.k8s.io/yaml"
 	"strings"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	apirequest "k8s.io/apiserver/pkg/endpoints/request"
 )
 
 func (mrt *manifestRoundTripper) getGroupResourceDiscovery(requestInfo *apirequest.RequestInfo) ([]byte, error) {
+	switch {
+	case requestInfo.Path == "/api":
+		ret, err := mrt.getAggregatedDiscoveryForURL("discovery-api.yaml", requestInfo.Path)
+		if os.IsNotExist(err) {
+			return mrt.getLegacyGroupResourceDiscovery(requestInfo)
+		}
+		return ret, err
+	case requestInfo.Path == "/apis":
+		ret, err := mrt.getAggregatedDiscoveryForURL("discovery-apis.yaml", requestInfo.Path)
+		if os.IsNotExist(err) {
+			return mrt.getLegacyGroupResourceDiscovery(requestInfo)
+		}
+		return ret, err
+	default:
+		// TODO can probably do better
+		return mrt.getLegacyGroupResourceDiscovery(requestInfo)
+	}
+}
+
+func (mrt *manifestRoundTripper) getAggregatedDiscoveryForURL(filename, url string) ([]byte, error) {
+	discoveryBytes, err := fs.ReadFile(mrt.sourceFS, filename)
+	if os.IsNotExist(err) {
+		discoveryBytes, err = fs.ReadFile(defaultDiscovery, filepath.Join("default-discovery", filename))
```

Comment:
won't this get us into trouble? The `applycfg` cmd will provide the input directory, which could contain data from the current cluster discovery.

Reply:
> won't this get us into trouble? The `applycfg` cmd will provide the input directory, which could contain data from the current cluster discovery.

The logic (I think) is, "prefer the must-gather discovery, if must-gather discovery doesn't exist, use a default".  So it's discovery of last resort.  It let's us work with older must-gathers and incomplete ones.